auth/credentials/oauth.go

```diff
@@ -11,41 +11,39 @@ import (
 	"net/url"
 	"strings"
 	"time"
-)
-
-// Token represents the internal OAuth2 Token structure
-type Token struct {
-	AccessToken string    `json:"access_token"`
-	Expiry      time.Time `json:"expiry,omitempty"`
-	ExpiresIn   int       `json:"expires_in"`
-}
 
-// TokenSource interface allows to fetch and revoke valid OAuth Access Tokens.
-type TokenSource interface {
```

Comment:
for simplicity you can do 
```go 
type TokenSource = oauth2.TokenSource
```

Reply:
SGTM. TokenSource is not exposed externally (only used internally). 
This line removes it. What will be the purpose of redefining token source in this case?

Reply:
it is, it's a public package with an exported name, if you really don't want people to use these types or methods then make them all unexported or `/internal` and we can move on

Reply:
We are exporting `RevocableTokenSource` in api.go. oauth2.Token source is only used as interface we inherit from.
That line removes TokenSource. Good point on not exposing only needed API. 
I will make oAuthTokenSource private as it is not needed to be exposed directly.